tests/test_integrated_channels/test_degreed/test_client.py

```diff
@@ -0,0 +1,287 @@
+            }]
+        }
+        degreed_api_client = DegreedAPIClient(self.enterprise_config)
+        output = degreed_api_client.delete_course_completion('fake-user', json.dumps(payload))
+
+        assert output == (200, '"{}"')
+        assert len(responses.calls) == 2
+        assert responses.calls[0].request.url == self.oauth_url
+        assert responses.calls[1].request.url == self.completion_status_url
+
+    @responses.activate
+    def test_create_course_content(self):
+        responses.add(
+            responses.POST,
+            self.oauth_url,
+            json=self.expected_token_response_body,
+            status=200
+        )
+        responses.add(
+            responses.POST,
+            self.course_url,
+            json='{}',
+            status=200
+        )
+
+        payload = {
+            'orgCode': 'org-code',
+            'providerCode': 'provider-code',
+            'courses': [{
+                'contentId': 'content-id',
+                'authors': [],
+                'categoryTags': [],
+                'url': 'url',
+                'imageUrl': 'image-url',
+                'videoUrl': 'video-url',
+                'title': 'title',
+                'description': 'description',
+                'difficulty': 'difficulty',
+                'duration': 20,
+                'publishDate': '2017-01-01',
+                'format': 'format',
+                'institution': 'institution',
+                'costType': 'paid',
+                'language': 'en'
+            }],
+        }
+        degreed_api_client = DegreedAPIClient(self.enterprise_config)
+        output = degreed_api_client.create_course_content(payload)
+        assert output == (200, '"{}"')
+        assert len(responses.calls) == 2
+        assert responses.calls[0].request.url == self.oauth_url
+        assert responses.calls[1].request.url == self.course_url
+
+    @responses.activate
+    def test_delete_course_content(self):
+        responses.add(
+            responses.POST,
+            self.oauth_url,
+            json=self.expected_token_response_body,
+            status=200
+        )
+        responses.add(
+            responses.DELETE,
+            self.course_url,
+            json='{}',
+            status=200
+        )
+
+        payload = {
+            'orgCode': 'org-code',
+            'providerCode': 'provider-code',
+            'courses': [{
+                'contentId': 'content-id',
+            }],
+        }
+        degreed_api_client = DegreedAPIClient(self.enterprise_config)
+        output = degreed_api_client.delete_course_content(payload)
+        assert output == (200, '"{}"')
+        assert len(responses.calls) == 2
+        assert responses.calls[0].request.url == self.oauth_url
+        assert responses.calls[1].request.url == self.course_url
+
+    @responses.activate
+    def test_expired_token(self):
+        """
+        If our token expires after some call, make sure to get it again.
```

Comment:
Well time is actually frozen for these tests, so if it expires in > 0 time it will never actually expire and we couldn't test token expiration. So in this test we're basically saying "Make a call, have the token expire after waiting some time (technically no time), and make a call again and notice 2 OAuth calls are required". And this contrasts with other tests where we don't need to make 2 OAuth calls, in particular the very next one, '`test_existing_token_is_valid`'.

Reply:
@UmanShahzad Ah, I see. I think it would be great to have this info available when reading the code, so how about putting it in the method's docstring?

Reply:
Sounds good, :+1: